configuration/exodus-lambda-deploy.yaml

```diff
@@ -0,0 +1,148 @@
+
+Parameters:
+  env:
+    Type: String
+    AllowedValues:
+      - dev
+      - stage
+      - prod
+    Default: dev
+    Description: The environment in which to deploy functions
+  oai:
+    Type: String
+    MinLength: 10
+    Description: The origin access identity ID associated with the environment
+
+Resources:
+  Distribution:
+    Type: AWS::CloudFront::Distribution
+    Properties:
+      DistributionConfig:
+        Comment: !Sub exodus-cdn-${env}
+        DefaultCacheBehavior:
+          AllowedMethods:
+            - GET
+            - HEAD
+          CachedMethods:
+            - GET
+            - HEAD
+          ForwardedValues:
+            QueryString: false
+            Cookies:
+              Forward: none
+          LambdaFunctionAssociations:
+            - EventType: origin-request
+              LambdaFunctionARN: !Ref OriginRequestFunc.Version
+            - EventType: origin-response
+              LambdaFunctionARN: !Ref OriginResponseFunc.Version
+          TargetOriginId: !Sub S3-exodus-cdn-${env}
+          ViewerProtocolPolicy: redirect-to-https
+        Enabled: true
+        HttpVersion: http2
+        Origins:
+          - DomainName: !Sub "exodus-cdn-${env}.s3.amazonaws.com"
+            Id: !Sub "S3-exodus-cdn-${env}"
+            S3OriginConfig:
+              OriginAccessIdentity: !Sub "origin-access-identity/cloudfront/${oai}"
+
+  OriginRequestFunc:
+    Type: AWS::Serverless::Function
+    Properties:
+      FunctionName: !Sub origin-request-${env}
+      CodeUri:
+      Handler: exodus_lambda.origin_request
+      Role: !GetAtt FuncRole.Arn
+      Runtime: python3.7
+      Timeout: 5
+      AutoPublishAlias: live
+
+  OriginResponseFunc:
+    Type: AWS::Serverless::Function
+    Properties:
+      FunctionName: !Sub origin-response-${env}
+      CodeUri:
+      Handler: exodus_lambda.origin_response
+      Role: !GetAtt FuncRole.Arn
+      Runtime: python3.7
+      Timeout: 5
+      AutoPublishAlias: live
+
+  FuncRole:
+    Type: AWS::IAM::Role
+    Properties:
+      RoleName: !Sub exodus-lambda-${env}
+      AssumeRolePolicyDocument:
+        Version: 2012-10-17
+        Statement:
+          - Effect: Allow
+            Action:
+              - sts:AssumeRole
+            Principal:
+              Service:
+                - lambda.amazonaws.com
+                - edgelambda.amazonaws.com
+      ManagedPolicyArns:
+        - arn:aws:iam::aws:policy/AWSLambdaFullAccess
+        - arn:aws:iam::aws:policy/CloudFrontFullAccess
+        - arn:aws:iam::aws:policy/AmazonS3ReadOnlyAccess
+        - arn:aws:iam::aws:policy/AWSCodePipelineFullAccess
+        - arn:aws:iam::aws:policy/AWSLambdaInvocation-DynamoDB
+      Path: /
+
+  OriginRequestAlarm:
+    Type: AWS::CloudWatch::Alarm
+    Properties:
+      AlarmName: !Sub origin-request-alarm-${env}
+      AlarmDescription: !Sub origin-request-${env} invocation errors
+      Namespace: AWS/Lambda
+      MetricName: Errors
```

Comment:
Do you know what kind of "errors" this means?  Is it only when the lambda fails to return a value (e.g. raises uncaught exception) or does it also include HTTP-level errors like a 404, which can happen even when lambda is working perfectly?

Reply:
I believe it's only concerned with errors raised by the code